src/js/pages/jobs/JobDetailPage.js

```diff
@@ -180,17 +182,14 @@ class JobDetailPage extends mixin(TabsMixin) {
     const job = this.props.job;
     const [schedule] = job.getSchedules();
 
-    const actions = [];
+    let actions = [];
 
     actions.push({
       label: "Edit",
       onItemSelect: this.props.handleEditButtonClick
     });
 
-    actions.push({
-      label: "Run Now",
-      onItemSelect: this.props.handleRunNowButtonClick
-    });
+    actions = actions.concat(menuGetActions(job.getId()));
```

Comment:
even if I like immutability a lot, only changing it for one line (but not above), isnt a good idea IMHO.

Also I dont really understand why you are extracting this into a new file, but most probably you'll add the otheres into that file?

Reply:
You are right, let that one slip. They should all be the immutable or not.

Reply:
By the way, it needs to be concat here, because menuGetActions returns a list. I will make everything concat.

Reply:
`menuGetActions(job.getId()).map(item => action.push(item))` :trollface: